### PR TITLE
Fix bug that allowed people to message someone they don't share a group with

### PR DESCRIPTION
**Pull request by tibetsprague**

### CODE REVIEW

File: api/models/post/findOrCreateThread.js [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
   if (!(participantIds && participantIds.length)) {
     throw new GraphQLYogaError("participantIds can't be empty")
   }
-  const checkForSharedGroup = id =>
-    Group.inSameGroup([userId, id])
-    .then(doesShare => {
-      if (!doesShare) throw new GraphQLYogaError(`no shared groups with user ${id}`)
-    })
-  return Promise.all(map(checkForSharedGroup, participantIds))
+  const validParticipantIds = await personFilter(userId)(User.where('id', 'in', uniq(participantIds))).fetchAll()
```
Comment (thomasgwatson) [inline api/models/post/findOrCreateThread.js:34]: Does this mean that everyone in a group chat needs to share a group?

Comment (thomasgwatson) [reply api/models/post/findOrCreateThread.js:34]: Maybe just the thread creator (looks at the test descriptions)

**Review by thomasgwatson — APPROVED** [review]
lgtm

**Comment by tibetsprague** [comment]
It means that the person creating the thread needs to share a group with everyone in the thread. On Jan 31, 2024, at 5:42 PM, Tom Watson ***@***.***> wrote:﻿
@thomasgwatson commented on this pull request.



In api/models/post/findOrCreateThread.js:
>    if (!(participantIds && participantIds.length)) {
     throw new GraphQLYogaError("participantIds can't be empty")
   }
-  const checkForSharedGroup = id =>
-    Group.inSameGroup([userId, id])
-    .then(doesShare => {
-      if (!doesShare) throw new GraphQLYogaError(`no shared groups with user ${id}`)
-    })
-  return Promise.all(map(checkForSharedGroup, participantIds))
+  const validParticipantIds = await personFilter(userId)(User.where('id', 'in', uniq(participantIds))).fetchAll()

Does this mean that everyone in a group chat needs to share a group?

**Pull request merged**